Added functional tests.

tmatsuo opened the pull request:
fixes #4.

tmatsuo commented:
@sjlangley 

Added some functional tests. I'd appreciate it if you could take a look.
The result is available at:
https://travis-ci.org/GoogleCloudPlatform/php-docker/builds/90852354

tmatsuo commented:
@marslangoog @bshaffer @SurferJeffAtGoogle  I'll appreciate your review too.

sjlangley commented:
Doesn't every file need the apache header?

tmatsuo commented:
Re: apache license headers, let's discuss on #8

If we want to @dlorenc to review, I'd add some backgrounds for these functional tests.

The PHP docker image provides features little bit complicated than images for other languages. For details, please see README.md at: https://github.com/GoogleCloudPlatform/php-docker/blob/master/php-nginx/README.md

In these tests, I want to make sure that

1. Some PHP functions are disabled by default and you can enable it by having php.ini in the project directory (phpinfo)
1. Some PHP extensions are compiled as shared lib and you need to explicitly enable it in php.ini (pdo_sqlite)
1. You can configure DOCUMENT_ROOT by setting an environment variable
1. You can add some pieces of nginx configuration file by having nginx-app.conf in your project directory (`/goodbye` URL only works with this)

While we also test that (these are already tested with the existing tests):
1. Docker build succeeds
1. Dynamically change the PHP version according to the contents in users' composer.json

tmatsuo commented:
Added the license headers. @sjlangley PTAL

sjlangley commented on testapps/php56/pdo_sqlite.php:
```diff
@@ -0,0 +1,19 @@
+ * limitations under the License.
+ */
+
+$pdo = new PDO('sqlite::memory:');
```
Shound't you echo 'PASS' here or something to test it executed as expected?

tmatsuo replied:
This app always fails :) The other app in php56_custom has some output.

sjlangley replied:
please add a comment above the statement that it is expected to fail

tmatsuo replied:
Done

sjlangley commented on testapps/php56_custom/web/phpinfo.php:
```diff
@@ -0,0 +1,18 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+phpinfo();
```
isn't phpinfo() disabled by default in the ini file?

tmatsuo replied:
Yes, I'd like to test the capability of enabling phpinfo by having a custom php.ini file in your project dir.

sjlangley commented on tests/ServingTest.php:
```diff
@@ -0,0 +1,68 @@
+    {
+        exec('docker run -d --name php56 -p 127.0.0.1:8080:8080 php56');
+        // Wait for nginx to start
+        sleep(3);
```
no better way to do this than sleep()?

tmatsuo replied:
Theoretically yes, but it works well for now. If it works reliably, I don't think it's warrant to spend engineering time. Can I revisit when the tests become flaky?

sjlangley commented on tests/CustomServingTest.php:
```diff
@@ -0,0 +1,75 @@
+    public function setUp()
+    {
+        exec('docker run -d --name php56_custom -p 127.0.0.1:8080:8080 '
+             .'php56_custom');
```
space after concat operator

tmatsuo replied:
Ok, I also added php-cs-fixer to check it.

tmatsuo commented:
Alright, added some comments, put spaces around concat operators, and added a script to check the syntax in the travis configuration file.

@sjlangley PTAL

sjlangley commented on tests/ServingTest.php:
```diff
@@ -0,0 +1,68 @@
+    private $pid;
+    private $client;
+
+    public function setUp()
```
Make this setupBeforeClass() so you can have > 1 test case easily?

tmatsuo replied:
Good point, done.

sjlangley commented on tests/ServingTest.php:
```diff
@@ -0,0 +1,68 @@
+                            . ' should be empty.');
+
+        // Access to pdo_sqlite.php, while the extention is not available.
+        try {
```
setExpectedException()?

tmatsuo replied:
Done

tmatsuo commented:
Thanks! Both done, PTAL @sjlangley

sjlangley commented:
lgtm++

sjlangley commented:
opps pressed wrong button

tmatsuo commented:
Thanks for the review! I'll merge this after the travis build succeeds. Also I'll update the image on php-mvm-a project too.

tmatsuo commented:
FYI, I have just updated the image on php-mvm-a and _manually_ test the deployment and web serving. It is required until we fix the issue #5.
